solutions/bronze/usaco-1013.mdx

```diff
@@ -113,6 +113,53 @@ public class swap {
 ```
 
 </JavaSection>
+<PySection>
+
+```py
+def reverse_segment(cows, start, end):
+	while start < end:
+		cows[start], cows[end] = cows[end], cows[start]
+		start += 1
+		end -= 1
+
+
+def find_cycle_length(cows, a1, a2, b1, b2):
+	# simulate the swap operation until we see a repeat
+	visited = set()
+	visited.add(tuple(cows))
+	while True:
+		reverse_segment(cows, a1 - 1, a2 - 1)
+		reverse_segment(cows, b1 - 1, b2 - 1)
+		if tuple(cows) in visited:
+			break
+		visited.add(tuple(cows))
+	# calculate the length of the cycle
+	cycle_len = len(visited)
+	return cycle_len
+
+
+def apply_swaps(cows, a1, a2, b1, b2, k):
+	cycle_len = find_cycle_length(cows, a1, a2, b1, b2)
+	k %= cycle_len
+	for i in range(k):
+		reverse_segment(cows, a1 - 1, a2 - 1)
+		reverse_segment(cows, b1 - 1, b2 - 1)
+
+	return cows
+
+
+n, k = map(int, input().split())
+a1, a2 = map(int, input().split())
+b1, b2 = map(int, input().split())
+
+cows = list(range(1, n + 1))
+cows = apply_swaps(cows, a1, a2, b1, b2, k)
+
+for cow in cows:
+	print(cow)
```

Comment:
wasn't this when you still had to print to a file?

Reply:
bump

Reply:
bump?

Reply:
not addressed

Reply:
what does bump mean?

Reply:
"bump" this up your priority list or something of the sort
not addressed, still

Reply:
I'm still not sure what you are asking for, when you say "wasn't this when you still had to print to a file?" what are you asking me to do in the code?

Reply:
your code uses stdin/stdout
shouldn't this problem require fin/fout?

Reply:
um I don't use stdin or stdout anywhere in my code

Reply:
you do? you read from input() & print w/ print()

Reply:
um is my last commit fine?